tests/unit/routers/test_schema_router.py

```diff
@@ -25,51 +24,80 @@ def test_schema_router_with_valid_json(test_client: TestClient) -> None:
     )
     assert response.status_code == status.HTTP_400_BAD_REQUEST
     assert response.json() == {
-        "detail": {"status": "error", "message": exception_messages.exception_400_invalid_version("current")}
+        "detail": {"status": "error", "message": exception_messages.exception_400_invalid_version("current")},
     }
 
 
 def test_schema_router_with_invalid_target_version(test_client: TestClient) -> None:
     """Test the post schema method with invalid target version."""
     response = test_client.post(
-        f"/schema?current_version={current_version}&target_version=2",
+        f"/schema?current_version={CURRENT_VERSION}&target_version=2",
         json={"valid_json": "valid_json"},
     )
     assert response.status_code == status.HTTP_400_BAD_REQUEST
     assert response.json() == {
-        "detail": {"status": "error", "message": exception_messages.exception_400_invalid_version("target")}
+        "detail": {"status": "error", "message": exception_messages.exception_400_invalid_version("target")},
     }
 
 
 def test_schema_router_with_empty_json(test_client: TestClient) -> None:
     """Test the post schema method with empty JSON."""
     response = test_client.post(
-        f"/schema?current_version={current_version}&target_version={target_version}",
+        f"/schema?current_version={CURRENT_VERSION}&target_version={TARGET_VERSION}",
         json={},
     )
     assert response.status_code == status.HTTP_400_BAD_REQUEST
     assert response.json() == {
-        "detail": {"status": "error", "message": exception_messages.EXCEPTION_400_EMPTY_INPUT_JSON}
+        "detail": {"status": "error", "message": exception_messages.EXCEPTION_400_EMPTY_INPUT_JSON},
     }
 
 
-def test_convert_schema_exception(test_client: TestClient) -> None:
-    """Test the convert schema method with an exception."""
-    SchemaProcessorService.convert_schema = MagicMock(side_effect=Exception("Test Exception"))
+def test_convert_schema_http_exception(test_client: TestClient) -> None:
+    """Test the convert schema method with an HTTPException."""
+    with patch(
+        "eq_cir_converter_service.services.schema.schema_processor_service.convert_schema",
+        side_effect=HTTPException(status_code=400, detail="Mocked HTTPException"),
+    ) as mock_convert_schema:
+        response = test_client.post(
+            f"/schema?current_version={CURRENT_VERSION}&target_version={TARGET_VERSION}",
+            json={"valid_json": "valid_json"},
+        )
+
+        assert response.status_code == status.HTTP_500_INTERNAL_SERVER_ERROR
+        assert response.json() == {
+            "detail": {"status": "error", "message": exception_messages.EXCEPTION_500_SCHEMA_PROCESSING},
+        }
+
+        mock_convert_schema.assert_called_with(
+            CURRENT_VERSION,
+            TARGET_VERSION,
+            {"valid_json": "valid_json"},
+        )
+
+
+def test_schema_router_with_matching_versions(test_client: TestClient) -> None:
```

Comment:
Only minor but I think potentially we could parametrize some of these tests to avoid repetition.